std-bits/table/src/main/java/org/enso/table/data/column/storage/DoubleStorage.java

```diff
@@ -102,14 +103,17 @@ private Storage fillMissingDouble(double arg) {
   }
 
   @Override
-  public Storage fillMissing(Object arg) {
-    if (arg instanceof Double) {
-      return fillMissingDouble((Double) arg);
-    } else if (arg instanceof Long) {
-      return fillMissingDouble((Long) arg);
-    } else {
-      return super.fillMissing(arg);
+  public Storage fillMissing(Value arg) {
+    if (arg.isNumber()) {
+      Object number = arg.as(Object.class);
```

Comment:
Rather than doing the conversion to object and then instance checking, maybe the "graal-way" of doing this would be to check if it `fitsInDouble`/`fitsInLong` instead?

Reply:
Correct. Please avoid `instanceof` check when dealing with `Value`. A `Value` can represent multiple things - for example a _date_ could represent both `asDate` as well as `asLong` (Milliseconds since the Era). That really requires calling `asXYZ` methods, as plain instances cannot handle both - date & long.

Reply:
I mostly wanted to preserve the existing logic - which was just expecting an Object and then doing instanceof checks. This logic is used all over the table library so I'd like to be consistent, we are using a Value here only to handle dates correctly.
Maybe we should adapt other places to also check values and not rely on instanceof but this feels like out of scope of this PR.